packages/gestalt/src/PageHeader.js

```diff
@@ -4,25 +4,83 @@ import Box from './Box.js';
+import styles from './PageHeader.css';
+
+export type ActionType = Element<
+  typeof Button | typeof IconButton | typeof Link | typeof Text | typeof Tooltip,
+>;
 
 type Props = {|
+  /**
+   * Add a badge displayed after the title. Be sure to localize the text.
+   */
+  badge?: {| text: string, tooltipText?: string |},
+  /**
+   * Specify a bottom border style for PageHeader: "sm" is 1px.
+   */
+  borderStyle?: 'sm' | 'none',
+  /**
+   * Label used for screen readers to provide information about the IconButton replacing actions the [sm breakpoint](https://gestalt.netlify.app/screen_sizes#Web-(px)).
+   */
+  dropdownAccessibilityLabel?: string,
+  /**
+   * Helper IconButton to be placed after the title for a supplemental Call To Action (CTA).
+   */
+  helperIconButton?: {|
+    accessibilityLabel: string,
+    accessibilityControls: string,
+    accessibilityExpanded: boolean,
+    onClick: ({|
+      event:
+        | SyntheticMouseEvent<HTMLButtonElement>
+        | SyntheticKeyboardEvent<HTMLButtonElement>
+        | SyntheticMouseEvent<HTMLAnchorElement>
+        | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+  /**
+   * Helper Link to be placed after the subtext.
+   */
+  helperLink?: {|
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+  /**
+   * Optional row of components.
+   */
+  items?: $ReadOnlyArray<Node>,
   /**
    * Use numbers for pixels: \`maxWidth={100}\` and strings for percentages: \`maxWidth="100%"\`. See the [max width](https://gestalt.pinterest.systems/pageheader#Max-width) variant for more info.
    */
   maxWidth?: Dimension,
   /**
-   * The primary action of the page. Can be [Button](https://gestalt.pinterest.systems/button), [Link](https://gestalt.pinterest.systems/link), [Tooltip](https://gestalt.pinterest.systems/tooltip) surrounding IconButton or a combination of IconButton, Tooltip and [Dropdown](https://gestalt.pinterest.systems/dropdown).
+   * The primary action of the page. Can be [Button](https://gestalt.pinterest.systems/button), [Link](https://gestalt.pinterest.systems/link), [Tooltip](https://gestalt.pinterest.systems/tooltip) surrounding IconButton or a combination of IconButton, Tooltip and [Dropdown](https://gestalt.pinterest.systems/dropdown). Primary and secondary actions are replaced with a [Dropdown](https://gestalt.netlify.app/dropdown) under the [sm breakpoint](https://gestalt.netlify.app/screen_sizes#Web-(px)). `primaryAction` takes both the main component and its equivalent using Dropdown subcomponents.
    */
-  primaryAction?: Element<typeof Button | typeof IconButton | typeof Link | typeof Tooltip>,
+  primaryAction?: {| component: ActionType, dropdownItems: $ReadOnlyArray<Node> |},
```

Comment:
I like this tighter coupling of the action and dropdown item. However: do we have a need to allow multiple dropdown items for a given action? Can we accept a Node rather than an array of Nodes for that field?

Reply:
If we are using IconButton as primary icon to open a Dropdown, we need an Array of Nodes to pass all the options in the Dropdown.
![Screen Shot 2022-04-04 at 3 37 43 PM](https://user-images.githubusercontent.com/10593890/161619086-6028d235-1517-43f9-a449-cea28742158d.png)

Reply:
Right, but that can be addressed where the Node(s) from the actions are passed to Dropdown. Is there a need for this field to accept an array? Is there a situation where the user needs to pass multiple dropdown items for a given action?